lib/Reader/readerir.cpp

```diff
@@ -945,6 +971,118 @@ void GenIR::createSafepointPoll() {
 
 bool GenIR::doTailCallOpt() { return JitContext->Options->DoTailCallOpt; }
 
+void GenIR::pushDebugScope(llvm::Function *F) {
+  LLILCDbgInfo.LexicalBlocks.push_back(LLILCDbgInfo.FnScopeMap[F]);
+}
+
+void GenIR::popDebugScope() {
+  LLILCDbgInfo.LexicalBlocks.pop_back();
+}
+
+// Set the Debug Location for the current instruction
+void GenIR::setDebugLocation(uint32_t CurrOffset, uint32_t IsCall) {
+  DIScope *Scope;
+  if (LLILCDbgInfo.LexicalBlocks.empty())
+    Scope = LLILCDbgInfo.TheCU;
+  else
+    Scope = LLILCDbgInfo.LexicalBlocks.back();
+  DebugLoc Loc = DebugLoc::get(CurrOffset, IsCall, Scope);
+
+  LLVMBuilder->SetCurrentDebugLocation(Loc);
+}
+
+llvm::DISubroutineType *GenIR::createFunctionType(llvm::Function *F,
```

Comment:
We have it because I needed something for the functiontype, and I wasn't sure if/how this was used throughout LLVM. I don't think it is necessary for us. I'm also not sure it's necessary for GDB for JIT (looking at the GDB EventListener)